app/models/journal.rb

```diff
@@ -59,19 +59,17 @@ class Journal < ActiveRecord::Base
   # Note for PostgreSQL: If this is called from inside a transaction, the lock will last until the
   #   end of that transaction.
   # Note for MySQL: THis method does not currently change anything (no locking at all)
-  def self.with_write_lock
+  def self.with_write_lock(journable)
+    lock_name =
     if OpenProject::Database.mysql?
-      Journal.transaction do
-        # MySQL is very weak when combining transactions and locks. Using an emulation layer to
-        # automatically release an advisory lock at the end of the transaction
-        TransactionalLock::AdvisoryLock.new('journals.write_lock').acquire
-        yield
-      end
+      # MySQL only supports a single lock
+      "journals.write_lock"
     else
-      Journal.transaction do
-        ActiveRecord::Base.connection.execute("LOCK TABLE #{table_name} IN SHARE ROW EXCLUSIVE MODE")
-        yield
-      end
+      "journal.#{journable.class}.#{journable.id}"
+    end
+
+    Journal.with_advisory_lock(lock_name, timeout_seconds: 60) do
```

Comment:
From the [docs](https://github.com/ClosureTree/with_advisory_lock#lock-wait-timeouts):

> Note that if a non-nil value is provided for timeout_seconds, the block will not be invoked if the lock cannot be acquired within that time-frame.

I tried it out and this really means that the block will silently not be executed if lock acquiring fails. This would mean then that the journal would not be added (again silently). Before, a failure to acquire the lock fails screaming (at least if I read to code and docs correctly of which I am not 100% sure). In any case, it should not fail silently.

The call to `with_advisory_lock` [returns false](https://github.com/ClosureTree/with_advisory_lock/blob/25e415c6f0b7626fa36726c524d1f32ebe217051/lib/with_advisory_lock/concern.rb#L11) if the lock could not be acquired so we should check for that. We only have to ensure that a non falsey value is returned from the block in case of a success.

Reply:
I suggest we explicitly use `with_advisory_lock_result` and raise if the lock wasn't acquired.